request.go

```diff
@@ -416,9 +416,12 @@ func unmarshalAttribute(
 		return
 	}
 
-	// JSON value was a float (numeric)
-	if value.Kind() == reflect.Float64 {
-		value, err = handleNumeric(attribute, fieldType, fieldValue)
+	switch attribute.(type) {
+	case json.Number:
+		value, err = handleNumeric(attribute.(json.Number), fieldType, fieldValue)
+		return
+	case float64:
+		value, err = handleNumeric(json.Number(fmt.Sprint(attribute)), fieldType, fieldValue)
```

Comment:
Not a fan of this `fmt.Sprint(attribute)`, but it does the job. A better alternatives is welcome.